OctoViewerTests/MockSafariPresenter.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  MockSafariPresenter.swift
+//  OctoViewer
+//
+//  Created by Hesham Salman on 5/26/17.
+//  Copyright © 2017 Hesham Salman
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+import Foundation
+@testable import OctoViewer
+
+class MockSafariPresenter: SafariPresenter {
+
+  var didOpenUrl = false
```

Comment:
You should make `didOpenUrl` an `Int` instead. You can also check if the method was called multiple times.